File: program/program/find-the-largest-two-elements-in-an-array/find_the_largest_two_elements_in_an_array.r
findLargestTwo <- function(arr) {
  # Check if the length of the array is less than 2
  if (length(arr) < 2) {
    cat("Array should have at least two elements.\n")
    return()
  }

  largest1 <- arr[1]
  largest2 <- arr[2]

  for (i in 3:length(arr)) {
    if (arr[i] > largest1) {
      largest2 <- largest1
      largest1 <- arr[i]
    } else if (arr[i] > largest2 && arr[i] != largest1) {
      largest2 <- arr[i]
    }
  }

  cat("The largest two elements are", largest1, "and", largest2, "\n")
}


arr1 <- c(12, 13, 1, 10, 34, 1)
findLargestTwo(arr1)

arr2 <- c(10, 5, 10)
findLargestTwo(arr2)

arr3 <- c(10, 10, 10)
findLargestTwo(arr3)

File: program/program/find-the-largest-two-elements-in-an-array/find_the_largest_two_elements_in_an_array.dart
import 'dart:io';

void findLargestTwo(List<int> arr) {
  if (arr.length < 2) {
    print("Array should have at least two elements.");
    return;
  }

  int largest = (arr[0] > arr[1]) ? arr[0] : arr[1];
  int secondLargest = (arr[0] > arr[1]) ? arr[1] : arr[0];

  for (int i = 2; i < arr.length; i++) {
    if (arr[i] > largest) {
      secondLargest = largest;
      largest = arr[i];
    } else if (arr[i] > secondLargest) {
      secondLargest = arr[i];
    }
  }

  print("Largest two elements in the array: $largest $secondLargest");
}

void main() {
  print("Enter integers for the array separated by space:");
  String input = stdin.readLineSync() ?? '';
  List<int> array = input.split(' ').map((e) => int.tryParse(e) ?? 0).toList();

  findLargestTwo(array);
}


/*

Command Prompt Execution:
dart find_the_largest_two_elements_in_an_array.dart
Enter integers for the array separated by space:
10 11 9 1 20 3
Largest two elements in the array: 20 11

*/
